src/components/SendTokens/ETHTokens.js

```diff
@@ -0,0 +1,495 @@
+      showModal: false,
+    });
+  };
+
+  handleChange = (value: Object) => {
+    const { activeAccount } = this.props;
+    // first update the amount, then after state is updated check for errors
+    this.setState({ value, gettingFee: true });
+    this.checkFormInputErrors();
+    if (activeAccount && checkIfSmartWalletAccount(activeAccount)) {
+      this.updateTxFee();
+      return;
+    }
+    const amount = parseFloat(get(value, 'amount', 0));
+    this.updateGasLimitAndTxFee(amount);
+  };
+
+  handleFormSubmit = async () => {
+    const { submitPressed } = this.state;
+    if (submitPressed) return;
+    this.formSubmitted = true;
+    this.setState({ submitPressed: true });
+    const value = this._form.getValue();
+    if (!value) return;
+    const { receiver, assetData, source } = this.props;
+    const txFeeInWei = await this.getTxFeeInWei();
+    // $FlowFixMe
+    let transactionPayload: TokenTransactionPayload = {
+      to: receiver,
+      amount: value.amount,
+      txFeeInWei,
+      symbol: assetData.token,
+      contractAddress: assetData.contractAddress,
+      decimals: assetData.decimals,
+    };
+    const { navigation, activeAccount } = this.props;
+    if (!activeAccount || !checkIfSmartWalletAccount(activeAccount)) {
+      const { gasLimit } = this.state;
+      const transactionSpeed = this.getTxSpeed();
+      const gasPrice = txFeeInWei.div(gasLimit).toNumber();
+      transactionPayload = {
+        ...transactionPayload,
+        gasPrice,
+        gasLimit,
+        txSpeed: transactionSpeed,
+      };
+    }
+    Keyboard.dismiss();
+    this.setState({ submitPressed: false }, () => {
+      navigation.navigate(SEND_TOKEN_CONFIRM, {
+        transactionPayload,
+        source,
+      });
+    });
+  };
+
+  useMaxValue = async () => {
+    const { balances, activeAccount, assetData } = this.props;
+    const { calculatingMaxValue } = this.state;
+    if (calculatingMaxValue) return;
+    this.setState({ calculatingMaxValue: true, gettingFee: true });
+    const { token } = assetData;
+    const balance = getBalance(balances, token);
+    const updatedState = {};
+    let txFeeInWei;
+    if (activeAccount && checkIfSmartWalletAccount(activeAccount)) {
+      txFeeInWei = await this.getSmartWalletTxFeeInWei(balance);
+    } else {
+      updatedState.gasLimit = await this.getGasLimit(balance); // calculate gas limit for max available balance
+      const transactionSpeed = this.getTxSpeed();
+      txFeeInWei = this.getTxFeeInWei(transactionSpeed, updatedState.gasLimit);
+    }
+    const maxAmount = calculateMaxAmount(token, balance, txFeeInWei);
+    const amount = formatAmount(maxAmount);
+    this.setState({
+      ...updatedState,
+      value: { amount },
+      gettingFee: false,
+      calculatingMaxValue: false,
+      txFeeInWei,
+    });
+    this.checkFormInputErrors();
+  };
+
+  getGasLimit = (amount: number) => {
+    // calculate either with amount in form or provided as param
```

Comment:
I'm sensing, that this comment is obsolete, as we are not taking amount from form data, just passing it as param